### PR TITLE
Looking For multiselect dropdown

### LINKED ISSUE #100: Project Form: Add multi select drop down for 'Looking for' - contributor type (PM, Front end.....)
#92 

**User story:** As a user, it is helpful to know the level of contributors project lead is seeking

**Deliverables**:
Add multi-select for contributor type, users should be allowed to input multiple values of their own contributor types.
E.G:
- Front end Engineer
- Product Managers
- Backend Engineers
- UX researcher

**Instructions:** To implement `Looking For` field use https://react-select.com/creatable same as in “Creatable Multiselect Example”.

**Mockup**
Screen Shot 2020-09-27 at 5 46 48 pm

Higher definition PDF file: 
[She's_Coding_Projects.pdf](https://github.com/shescoding/projects-platform-frontend/files/5289260/She.s_Coding_Projects.pdf)

### PR DESCRIPTION
**Pull request by nadyafebi**
Change the Looking For field of the Project Form to use Creatable from `react-select`. Closes #100.

**Changes**
* Added `react-select` as dependency.
* Change the field to use `Creatable` component.
* Added new function `handleCreatableChange` to help handle changes from `Creatable` component.
* Modified the `validateForm` function to handle `null` value.
* Modified the `handleSubmit` function to format the field value from array to comma-separated string.
* Style the field as best as I can to look similar to the other fields.

**Demo**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "react-modal": "3.11.2",
     "react-router-dom": "5.1.2",
     "react-scripts": "3.2.0",
+    "react-select": "^3.1.0",
```
Comment (nurmerey) [inline package.json:20]: Please remove `^`. This ensures that we lock in the library version and don't accidentally cause any issues down the line.

File: src/pages/AddProject.tsx [hunk 2]
```diff
@@ -66,6 +74,25 @@ class AddProject extends React.Component<RouteComponentProps, FormState> {
     cancelModalOpen: false
   }
 
+  lookingForOptions: FormOptions = [
```
Comment (nurmerey) [inline src/pages/AddProject.tsx:77]: Let's add a comment at the top that these values will be coming in from a getPosition API when #114 is implemented.

**Review by nurmerey — CHANGES_REQUESTED** [review]
Asked for a few minor changes! Otherwise LGTM 🙌  ! Thank you @nadyafebi

**Comment by nadyafebi** [comment]
@nurmerey I changed the stuff you mentioned 😄

File: yarn.lock [hunk 3]
```diff
@@ -906,6 +906,13 @@
   dependencies:
     regenerator-runtime "^0.13.2"
 
+"@babel/runtime@^7.4.4", "@babel/runtime@^7.5.5", "@babel/runtime@^7.7.2", "@babel/runtime@^7.8.7":
```
Comment (nurmerey) [inline yarn.lock:909]: `yarn.lock` should be in your `.gitignore` file, and should not be committed. Can you please remove this file from this PR? Everything else looks good!

**Review by nurmerey — CHANGES_REQUESTED** [review]
One more ask, to remove `yarn.lock` file. Thank you!

**Comment by nadyafebi** [comment]
@nurmerey Removed the yarn.lock (not sure how it got committed there) and also resolved some merge conflict 😃

**Comment by nadyafebi** [comment]
Hi @nurmerey @ctran14, I saw someone else got assigned to issue #100. Should I just close this PR then? Thanks

**Comment by nurmerey** [comment]
> Hi @nurmerey @ctran14, I saw someone else got assigned to issue #100. Should I just close this PR then? Thanks

Hey sorry for the mixup, the PR looks good. I will approve and merge it (and reassign the task for you!).

**Review by nurmerey — APPROVED** [review]
LGTM 💐 Thank you for great work!

**Pull request merged**